src/uu/seq/src/seq.rs

```diff
@@ -268,38 +200,64 @@ fn done_printing<T: Num + PartialOrd>(next: &T, increment: &T, last: &T) -> bool
     }
 }
 
-/// Floating point based code path
+/// Print a sequence of floating point numbers.
+///
+/// This function prints a sequence of floating point numbers. The
+/// first, last, and increment are given in the triple `range`. Each is
+/// a [`BigDecimal`]. The `separator` is printed between the elements of
+/// the sequence and `terminator` after the last element.
+///
+/// `largest_dec` gives the number of digits of precision to use when
+/// printing the digits after the decimal point, padded with zeros on
+/// the right. If `pad` is true, then `padding` is used to determine the
+/// number of digits to print before the decimal point, padded with
+/// zeros on the left.
+///
+/// If `is_first_minus_zero` is `true`, then the `first` parameter is
+/// printed as if it were negative zero. Only set this to `true` if
+/// `first` is actually zero.
 fn print_seq(
-    range: RangeF64,
+    range: RangeBigDecimal,
     largest_dec: usize,
     separator: String,
     terminator: String,
     pad: bool,
     padding: usize,
-) {
+    is_first_minus_zero: bool,
+) -> std::io::Result<()> {
+    let stdout = stdout();
+    let mut stdout = stdout.lock();
     let (first, increment, last) = range;
-    let mut i = 0isize;
-    let mut value = first + i as f64 * increment;
+    let mut value = first.clone();
+    let mut is_first_iteration = true;
     while !done_printing(&value, &increment, &last) {
+        let mut width = padding;
+        if is_first_iteration && is_first_minus_zero {
+            print!("-");
+            width -= 1;
+        }
+        is_first_iteration = false;
         let istr = format!("{:.*}", largest_dec, value);
         let ilen = istr.len();
         let before_dec = istr.find('.').unwrap_or(ilen);
-        if pad && before_dec < padding {
-            for _ in 0..(padding - before_dec) {
-                print!("0");
+        if pad && before_dec < width {
+            for _ in 0..(width - before_dec) {
+                write!(stdout, "0")?;
             }
         }
-        print!("{}", istr);
-        i += 1;
-        value = first + i as f64 * increment;
+        write!(stdout, "{}", istr)?;
```

Comment:
This seems to do the trick:
```suggestion
        write!(
            stdout,
            "{value:>0width$.precision$}",
            value = value,
            width = if pad { width + 1 + largest_dec } else { 0 },
            precision = largest_dec
        )?;
```

Reply:
I'll put your suggested change in a separate pull request.

Reply:
https://github.com/uutils/coreutils/pull/2676